src/pages/index.js

```diff
@@ -97,10 +102,113 @@ const newCardPopup = new PopupWithForm(
 );
 newCardPopup.setEventListeners();
 
-const userInfo = new UserInfo(".profile__title", ".profile__description");
+//Profile creation and editing
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api.getUser().then((inputValues) => {
+  userInfo.setUserInfo(inputValues.name, inputValues.about);
+  userInfo.setAvatar(inputValues.avatar);
+});
+
+function handleProfileEditSubmit(inputValues) {
+  profileEditPopup.renderLoading(true);
+  api
+    .editProfile(inputValues.name, inputValues.about)
+    .then((data) => {
+      userInfo.setUserInfo(data.name, data.about);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error occurred while updating profile", err);
+    })
+    .finally(() => {
+      profileEditPopup.renderLoading(false);
+      profileEditPopup.close();
+    });
+}
+
+profileEditButton.addEventListener("click", () => {
+  const data = userInfo.getUserInfo();
+  profileTitleInput.value = data.name;
+  profileDescriptionInput.value = data.about;
+  profileEditPopup.open();
+});
+
+//DELETING CARDS
 
 const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
 profileEditPopup.setEventListeners();
+
+const deleteCardPopup = new PopupConfirmDelete(
+  "#delete-card-modal",
+  handleDeleteCard
+);
+
+function handleDeleteCard(cardID) {
+  deleteCardPopup.open();
+  deleteCardPopup.confirmDelete(cardID);
+  api.deleteRequest(cardID);
```

Comment:
You need to **delete** cards only in block `then` after a **successful** response from the server so that if there is an error the card would stay in the `DOM`.